### PR TITLE
misc improvements and updateIndexes introduction.

### PR DESCRIPTION
**Pull request by elcerdo**

### CODE REVIEW

**Comment by elcerdo** [comment]
not much to say. pretty straightforward.

File: ChangeLog.md [hunk 1]
```diff
@@ -42,6 +42,17 @@
    (Pierre Gueth [#1008](https://github.com/DGtal-team/DGtal/pull/1008))
  - Mutable iterator on DiscreteExteriorCalculus.
    (Pierre Gueth [#1008](https://github.com/DGtal-team/DGtal/pull/1008))
+ - Unary minus operators for k-forms, vector fields and linear operators.
+   (Pierre Gueth [#1020](https://github.com/DGtal-team/DGtal/pull/1020))
+ - Introduction of .updateIndexes() that need to be called after any
```
Comment (dcoeurjo) [inline ChangeLog.md]: that needs

File: src/DGtal/dec/DiscreteExteriorCalculus.h [hunk 2]
```diff
@@ -294,13 +316,42 @@ namespace DGtal
...
+     * @tparam order order of signed cells.
+     * @tparam duality duality of signed cells.
+     * @return index ordered signed cells.
+     */
+    template <Order order, Duality duality>
+    const SCells&
+    getIndexedSCells() const;
+
+    /**
+     * Reorder operator from order-forms to order-forms.
```
Comment (dcoeurjo) [inline src/DGtal/dec/DiscreteExteriorCalculus.h]: from ordered-forms to ordered-forms ?

File: src/DGtal/dec/doc/moduleDECIntroduction.dox [hunk 3]
```diff
@@ -245,17 +245,29 @@ corresponding factory type.
...
-implementing the CDigitalSet concept using
-DiscreteExteriorCalculusFactory::createFromDigitalSet. This factory static
-member inserts each point as a primal n-cell in the structure and then fills
-in between k-cells to glue all n-cells together. Borders can be added or
-removed from the generated structure using the add_border boolean.
+DiscreteExteriorCalculus objects can also be created
+using the DiscreteExteriorCalculusFactory factory.
+DiscreteExteriorCalculusFactory::createFromDigitalSet static
+member inserts each point of a digital set as a primal n-cell in the structure and then fills
+in between k-cells to glue all n-cells together.
```
Comment (dcoeurjo) [inline src/DGtal/dec/doc/moduleDECIntroduction.dox]: in between -> in-between

File: src/DGtal/dec/doc/moduleDECIntroduction.dox [hunk 4]
```diff
@@ -245,17 +245,29 @@ corresponding factory type.
...
+to enforce various boundary condition.
 Dual k-cells are created automatically with the right size ratio.
 Note that the structure can be altered after creation to adjust small details
-(DiscreteExteriorCalculus.insertSCell and DiscreteExteriorCalculus.eraseCell),
-but it should not be modified while any KForm, VectorField or LinearOperator are instancied.
-See \ref moduleDECEmbedding for a discussion about manifold embedding.
+or created completely manually via calls to
+DiscreteExteriorCalculus.insertSCell and DiscreteExteriorCalculus.eraseCell.
+
+@note Internal indexes should rebuild by calling DiscreteExteriorCalculus.updateIndexes
```
Comment (dcoeurjo) [inline src/DGtal/dec/doc/moduleDECIntroduction.dox]: should rebuild -> should be reset

File: src/DGtal/dec/doc/moduleDECIntroduction.dox [hunk 5]
```diff
@@ -266,6 +278,7 @@ coordinates of such cells are given in the same system as points from the
 input set.
 The second call removes a primal 1-cell and opens the hole created by the first call.
 Here coordinates are given in the KhalimskySpaceND signed cell frame of reference.
+To finalize the structure, a call to DiscreteExteriorCalculus.updateIndexes is made.
```
Comment (dcoeurjo) [inline src/DGtal/dec/doc/moduleDECIntroduction.dox]: is performed

File: src/DGtal/io/Display2DFactory.ih [hunk 6]
```diff
@@ -179,7 +179,7 @@ DGtal::Display2DFactory::drawWithColorMap(DGtal::Board2D& board, const DGtal::KF
 {
     typedef typename TCalculus::Scalar Scalar;
     typedef typename TCalculus::SCell SCell;
-    typedef typename TCalculus::KSpace KSpace;
+    //typedef typename TCalculus::KSpace KSpace;
```
Comment (dcoeurjo) [inline src/DGtal/io/Display2DFactory.ih]: cleanup

File: src/DGtal/io/Display3DFactory.ih [hunk 7]
```diff
@@ -105,7 +105,7 @@ DGtal::Display3DFactory<Space, KSpace>::drawWithColorMap(Display3D<Space, KSpace
     ASSERT( kform.myCalculus );
 
     typedef typename Calculus::Scalar Scalar;
-    typedef typename Calculus::KSpace KSpace;
+    //typedef typename Calculus::KSpace KSpace;
```
Comment (dcoeurjo) [inline src/DGtal/io/Display3DFactory.ih]: cleanup

File: src/DGtal/dec/DiscreteExteriorCalculus.h (hunk 2 above)
Comment (elcerdo) [inline src/DGtal/dec/DiscreteExteriorCalculus.h]: see other memebers description (ie identity or derivative).
order refers to the template parameter.

Comment (dcoeurjo) [inline src/DGtal/dec/DiscreteExteriorCalculus.h]: ok I see.. may I suggest ```<order>-form``` instead ?

Comment (elcerdo) [inline src/DGtal/dec/DiscreteExteriorCalculus.h]: or k-form of order order

Comment (dcoeurjo) [inline src/DGtal/dec/DiscreteExteriorCalculus.h]: (order)-form or <order>-form

File: src/DGtal/dec/DiscreteExteriorCalculus.h [hunk 8]
```diff
@@ -294,14 +316,43 @@ namespace DGtal
...
+     * @tparam order order of signed cells.
+     * @tparam duality duality of signed cells.
+     * @return index ordered signed cells.
+     */
+    template <Order order, Duality duality>
+    const SCells&
+    getIndexedSCells() const;
+
+    /**
+     * Reorder operator from <em>order</em>-forms to <em>order</em>-forms.
```
Comment (dcoeurjo) [inline src/DGtal/dec/DiscreteExteriorCalculus.h]: <em> is HTML only. Please use @a or @e in doxygen

Comment (elcerdo) [inline src/DGtal/dec/DiscreteExteriorCalculus.h]: i can't get the @e command to work on something like (order-1)-form...
doxygen doc says "To emphasize multiple words use <em>multiple words</em>." without limiting its use to html or latex output.

Comment (dcoeurjo) [inline src/DGtal/dec/DiscreteExteriorCalculus.h]: just to @e for each word, or used markdown syntax (cf doxygen website)

Comment (elcerdo) [inline src/DGtal/dec/DiscreteExteriorCalculus.h]: switched to markdown syntax.

**Comment by dcoeurjo** [comment]
Many thanks
Merging

**Pull request merged**